tests/run-make/extern-flag-disambiguates/rmake.rs

```diff
@@ -0,0 +1,30 @@
+//@ ignore-cross-compile
+
+use run_make_support::{cwd, run, run_in_tmpdir, rustc};
+
+// Attempt to build this dependency tree:
+//
+//	A.1   A.2
+//	 |\    |
+//	 | \   |
+//        B  \  C
+//         \ | /
+//          \|/
+//           D
+//
+// Note that A.1 and A.2 are crates with the same name.
+
+fn main() {
+    run_in_tmpdir(|| {
+        rustc().metadata("1").extra_filename("-1").input("a.rs").run();
+        rustc().metadata("2").extra_filename("-2").input("a.rs").run();
+        rustc().input("b.rs").extern_("a", "liba-1.rlib").run();
```

Comment:
There is `run_make_support::rust_lib_name` for generating the `rlib` filename.

Reply:
that is neat, but i'm not entirely sure the advantage over just using string literals, it's more typing and it's arguably less readable due to adding more indirection.

Reply:
Personally, I use it because it highlights how it's the compiled version of `a.rs` with its extra filename appended:

`rust_lib_name(a-1)`

Otherwise, I read "liba" first forgetting the "lib" prefix and wonder for a split second what it's referring to.

I agree that it's a lot less important than the `static_lib_name` or `dynamic_lib_name` counterparts which actually adjust themselves depending on the current platform, though.

Reply:
I agree that for `rlib` specifically, it's a bit noisy, but we already use it everywhere, so this keeps it consistent. And it also allows us to grep for all rlib usages within rmake tests easily.